raw.go

```diff
@@ -31,3 +31,32 @@ type RawSpan struct {
 	// The span's associated baggage.
 	Baggage map[string]string // initialized on first use
 }
+
+// SpanDetail summarizes the identifying fields of a Span, for
+// diagnostic and implementation-specific purposes.
+type SpanDetail struct {
```

Comment:
This approach: N+K fields are copied, there is one new struct definition, one new method definition, one type assertion, and one method call.
That approach: K fields are copied once, there is one new interface definition, N new method definitions, one type assertion, and K method calls.

(where K (number of fields accessed) is <= N (number of detail fields))

I'm not sure there is a significant difference either way, but it feels more straightforward to me this way. In our use-case, we have N=3 (Context, Operation, Start) and K=2 (Context, Start), fwiw.